src/main/java/hudson/plugins/audit_trail/LogFileAuditLogger.java

```diff
@@ -30,33 +48,143 @@ public class LogFileAuditLogger extends AuditLogger {
 
     private static final Logger LOGGER = Logger.getLogger(LogFileAuditLogger.class.getName());
     static final String DEFAULT_LOG_SEPARATOR=" ";
+    protected static final String DAILY_ROTATING_FILE_REGEX_PATTERN = "-[0-9]{4}-[0-9]{2}-[0-9]{2}" + ".*" + "(?<!lck)$";
     @Nonnull
     private String logSeparator;
 
+    /**
+     * If true, enables the daily rotation of the logs. Limit will be configure with a 0 value
+     * (unlimitted size), and count will be set-up as 1 for FileHandler.
+     * @see #getRotateDaily()
+     */
+    public boolean rotateDaily;
+
     private transient FileHandler handler;
+    private transient ZonedDateTime initInstant;
+    private transient Path basePattern;
+    private transient final Object monitor = new Object();
 
-    @DataBoundConstructor
+    @Deprecated
     public LogFileAuditLogger(String log, int limit, int count, String logSeparator) {
+        this(log, limit, count, logSeparator, false);
+    }
+
+    @DataBoundConstructor
+    public LogFileAuditLogger(String log, int limit, int count, String logSeparator, boolean rotateDaily) {
         this.log = Util.replaceMacro(log, EnvVars.masterEnvVars);
-        this.limit = limit;
+        this.basePattern = Paths.get(log);
+        this.limit = rotateDaily ? 0 : limit;
```

Comment:
Useless since you do not read the `limit` in the `configure` method.
```suggestion
        this.limit = limit;
```